esp-hal-common/src/systimer.rs

```diff
@@ -225,19 +225,20 @@ impl<const CHANNEL: u8> Alarm<Target, CHANNEL> {
 }
 
 impl<const CHANNEL: u8> Alarm<Periodic, CHANNEL> {
-    pub fn set_period(&self, period: fugit::HertzU32) {
-        let time_period: MicrosDurationU32 = period.into_duration();
-        let us = time_period.ticks();
+    pub fn set_period(&self, period: MicrosDurationU32) {
```

Comment:
I guess a duration better matches a time period but just want to make sure we all agree since it's a change in public API

Reply:
I would recommend changing the name otherwise, it was misleading before IMO. Don't really care which units we ultimately accept as arguments but for the given function name this made more sense to me.